platform-sdk/swirlds-common/src/main/java/com/swirlds/common/merkle/synchronization/TeachingSynchronizer.java

```diff
@@ -91,24 +92,18 @@ public class TeachingSynchronizer {
     /**
      * Create a new teaching synchronizer.
      *
-     * @param threadManager
-     * 		responsible for managing thread lifecycles
-     * @param in
-     * 		the input stream
-     * @param out
-     * 		the output stream
-     * @param root
-     * 		the root of the tree
-     * @param breakConnection
-     * 		a method that breaks the connection. Used iff
-     * 		an exception is encountered. Prevents deadlock
-     * 		if there is a thread stuck on a blocking IO
-     * 		operation that will never finish due to a
-     * 		failure.
-     * @param reconnectConfig
-     *      reconnect configuration from platform
+     * @param platformContext the platform context
+     * @param threadManager   responsible for managing thread lifecycles
+     * @param in              the input stream
+     * @param out             the output stream
+     * @param root            the root of the tree
+     * @param breakConnection a method that breaks the connection. Used iff an exception is encountered. Prevents
+     *                        deadlock if there is a thread stuck on a blocking IO operation that will never finish due
+     *                        to a failure.
+     * @param reconnectConfig reconnect configuration from platform
      */
     public TeachingSynchronizer(
+            @NonNull final PlatformContext platformContext,
```

Comment:
Let's try to avoid using PlatformContext in common

Reply:
The reconnect classes don't really belong in common. But since I don't want to add 5000 lines to this PR, simpler just to do as you ask and remove the context.